tools/tm-bench/main.go

```diff
@@ -189,6 +189,16 @@ func calculateStatistics(
 		offset = len(blockMetas)
 	}
 
+	// Ignore the first block created if there were no txs in it, since we likely didn't begin
```

Comment:
Thats a good point! In the case of multiple connections should we set it as the point at which all connections have begun? (I think so, since thats the point at which we've met the contract promised by tm-bench)

Reply:
before we send 1st transaction is fine, I guess

Reply:
I've implemented that (See latest commit), by making the start time after all the send and receive loops have begun. However this isn't working as expected, heres some of the log from verbose mode:
```
valar@Valar-Ubuntu:~/Code/go/src/github.com/tendermint/tendermint/tools/tm-bench$ tm-bench -v -r 100000 localhost:26657
<snip>
D[07-12|04:53:19.059] 5990 txs in block 0, height 7666             
D[07-12|04:53:19.059] 7251 txs in block 1, height 7665             
D[07-12|04:53:19.059] 7369 txs in block 2, height 7664             
D[07-12|04:53:19.059] 6852 txs in block 3, height 7663             
D[07-12|04:53:19.059] 7106 txs in block 4, height 7662             
D[07-12|04:53:19.059] 4989 txs in block 5, height 7661             
D[07-12|04:53:19.059] 757 txs in block 6, height 7660              
D[07-12|04:53:19.060] 0 txs in block 7, height 7659 
```
Redoing multiple times, sometimes the first block is non-zero, but its always lower than the rest. I'm not sure whats going on to make the second to last block have a low tx count. (Perhaps the send loops haven't actually started firing)

Reply:
Interesting. Worth investigating futher

Reply:
Confirmed, send loops didn't actually start firing at that time